vm/cli/commands/run.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+pub const LOCALE: &num_format::Locale = &num_format::Locale::en;
+
+/// Executes an Aleo program function locally
+#[derive(Debug, Parser)]
+pub struct Run {
+    /// The function name.
+    #[clap(parse(try_from_str))]
+    function: Identifier<CurrentNetwork>,
+    /// The function inputs.
+    #[clap(parse(try_from_str))]
+    inputs: Vec<Value<CurrentNetwork>>,
+    /// Uses the specified endpoint.
+    #[clap(long)]
+    endpoint: Option<String>,
+    /// Toggles offline mode.
+    #[clap(long)]
+    offline: bool,
+}
+
+impl Run {
+    /// Compiles an Aleo program function with the specified name.
+    #[allow(clippy::format_in_format_args)]
+    pub fn parse(self) -> Result<String> {
+        // Derive the program directory path.
+        let path = std::env::current_dir()?;
+
+        // Load the package.
+        let package = Package::open(&path)?;
```

Comment:
I think the design of the package's implementation of package.json is currently flawed. It has a private_key in the .json file and people are using are CLIs in production with this private key stored in their clouds. Given we will have custodial solutions being built around Aleo, this is a security nightmare. We shouldn't have the private key stored in a plaintext file and instead enforce a more secure way of inserting the private key for local program execcution. I think we should address this before making a public release of the API anywhere.

Reply:
This private key is only meant to be used for development (as its name implies), it shouldn't be for production. I agree we should patch this, let's prioritize this during Phase 3 (read: before mainnet).